depscan/cli.py

```diff
@@ -742,8 +742,24 @@ def main():
             LOG.debug("Vulnerability database loaded from %s", config.vdb_bin_file)
 
         sources_list = [OSVSource(), NvdSource()]
-        if os.environ.get("GITHUB_TOKEN"):
-            sources_list.insert(0, GitHubSource())
+        github_token = os.environ.get("GITHUB_TOKEN")
+        if github_token:
+            github_client = github.GitHub(github_token)
+            
+            if not github_client.authenticate():
+                LOG.error("The GitHub personal access token supplied appears to be invalid or expired. Please see: https://github.com/owasp-dep-scan/dep-scan#github-security-advisory")
+            elif github_client.get_token_scopes() is None:
```

Comment:
Is depscan not working with fine grained tokens?

Reply:
I saw [this note](https://docs.github.com/en/rest/packages/packages?apiVersion=2022-11-28#about-github-packages) in GitHub documentation for the packages API endpoint: `you must authenticate using a personal access token (classic)`.

Reply:
This needs to be tested as well to prevent users from getting stuck with a legacy token.